src/Bindings/LuaState.h

```diff
@@ -263,6 +266,19 @@ class cLuaState
 	bool GetStackValue(int a_StackPos, eBlockFace & a_Value);
 	bool GetStackValue(int a_StackPos, eWeather & a_Value);
 	bool GetStackValue(int a_StackPos, float & a_ReturnedVal);
+	bool GetStackValue(int a_StackPos, int & a_Value);
+
+	// GetUserType() retrieves the user type at a_StackPos, if it is a user type of a_Type.
+	// If not, a_UserType remains unchanged.
+	template <typename T>
+	bool GetUserType(int a_StackPos, T * & a_UserType, const char * a_Type)
+	{
+		if (CheckParamUserType(a_StackPos, a_Type))
+		{
+			a_UserType = static_cast<T *>(tolua_tousertype(m_LuaState, a_StackPos, nullptr));
+		}
+		return false;
+	}
```

Comment:
This is not needed. Client code should first check all parameters, and then use `GetStackValue()` or `GetStackValues()`.

I'm considering a rewrite of the `CheckParamXYZ` handling so that it would better support overloaded functions, and for that I need the current usage idiom.